src/mongo_logger.ts

```diff
@@ -124,56 +176,247 @@ function resolveLogPath(
+          ev.options;
+        log = {
+          ...log,
+          maxIdleTimeMS,
+          minPoolSize,
+          maxPoolSize,
+          maxConnecting,
+          waitQueueTimeoutMS
+        };
+        log.waitQueueSize = ev.waitQueueSize;
+      }
+      break;
+    case CONNECTION_POOL_READY:
+      ev = logObject as ConnectionPoolReadyEvent;
+      log = attachConnectionFields(log, ev);
+      log.message = 'Connection pool ready';
+      break;
+    case CONNECTION_POOL_CLEARED:
+      ev = logObject as ConnectionPoolClearedEvent;
+      log = attachConnectionFields(log, ev);
+      log.message = 'Connection pool cleared';
+      if (ev.serviceId?._bsontype === 'ObjectId') {
+        log.serviceId = ev.serviceId.toHexString();
+      }
+      break;
+    case CONNECTION_POOL_CLOSED:
+      ev = logObject as ConnectionPoolClosedEvent;
+      log = attachConnectionFields(log, ev);
+      log.message = 'Connection pool closed';
+      break;
+    case CONNECTION_CREATED:
+      ev = logObject as ConnectionCreatedEvent;
+      log = attachConnectionFields(log, ev);
+      log.message = 'Connection created';
+      log.driverConnectionId = ev.connectionId;
+      break;
+    case CONNECTION_READY:
+      ev = logObject as ConnectionReadyEvent;
+      log = attachConnectionFields(log, ev);
+      log.message = 'Connection ready';
+      log.driverConnectionId = ev.connectionId;
+      break;
+    case CONNECTION_CLOSED:
+      ev = logObject as ConnectionClosedEvent;
+      log = attachConnectionFields(log, ev);
+      log.message = 'Connection closed';
+      log.driverConnectionId = ev.connectionId;
+      log.reason = ev.reason;
+      if (ev.reason === 'error') {
+        // TODO: Set log.error
+        // log.error = ev
+      }
+      break;
+    case CONNECTION_CHECK_OUT_STARTED:
+      ev = logObject as ConnectionCheckOutStartedEvent;
+      log = attachConnectionFields(log, ev);
+      log.message = 'Connection checkout started';
+      break;
+    case CONNECTION_CHECK_OUT_FAILED:
+      ev = logObject as ConnectionCheckOutFailedEvent;
+      log = attachConnectionFields(log, ev);
+      log.message = 'Connection checkout failed';
+      log.reason = ev.reason;
+      break;
+    case CONNECTION_CHECKED_OUT:
+      ev = logObject as ConnectionCheckedOutEvent;
+      log = attachConnectionFields(log, ev);
+      log.message = 'Connection checked out';
+      log.driverConnectionId = ev.connectionId;
+      break;
+    case CONNECTION_CHECKED_IN:
+      ev = logObject as ConnectionCheckedInEvent;
+      log = attachConnectionFields(log, ev);
+      log.message = 'Connection checked in';
+      log.driverConnectionId = ev.connectionId;
+      break;
+    default:
+      for (const key in logObject) {
+        const value = logObject[key];
+        // eslint-disable-next-line no-restricted-syntax
+        if (value === undefined || value === null) continue;
+        log[key] = value;
+      }
```

Comment:
Is there a better way to do this? The spec requires that for null or undefined fields, we delete the key itself, so optional chaining doesn't satisfy our requirements here.